packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -82,1566 +70,1794 @@ const GOERLI = {
   ticker: NetworksTicker.goerli,
 };
 
-const controllerName = 'TokensController' as const;
-
 describe('TokensController', () => {
-  let tokensController: TokensController;
-  let approvalController: ApprovalController;
-  let messenger: ControllerMessenger<
-    AllowedActions,
-    AllowedEvents | ApprovalControllerEvents
-  >;
-  let tokensControllerMessenger;
-  let approvalControllerMessenger;
-  let getNetworkClientByIdHandler: jest.Mock<
-    ReturnType<NetworkController['getNetworkClientById']>,
-    Parameters<NetworkController['getNetworkClientById']>
-  >;
-
-  const changeNetwork = (providerConfig: ProviderConfig) => {
-    messenger.publish(`NetworkController:networkDidChange`, {
-      ...defaultNetworkState,
-      providerConfig,
-    });
-  };
-
-  const triggerPreferencesStateChange = (state: PreferencesState) => {
-    messenger.publish('PreferencesController:stateChange', state, []);
-  };
-
-  const fakeProvider = new FakeProvider();
-
-  beforeEach(async () => {
-    const defaultSelectedAddress = '0x1';
-    messenger = new ControllerMessenger();
-
-    approvalControllerMessenger = messenger.getRestricted({
-      name: 'ApprovalController',
-      allowedActions: [],
-      allowedEvents: [],
-    });
-
-    tokensControllerMessenger = messenger.getRestricted<
-      typeof controllerName,
-      AllowedActions['type'],
-      AllowedEvents['type']
-    >({
-      name: controllerName,
-      allowedActions: [
-        'ApprovalController:addRequest',
-        'NetworkController:getNetworkClientById',
-      ],
-      allowedEvents: [
-        'NetworkController:networkDidChange',
-        'PreferencesController:stateChange',
-        'TokenListController:stateChange',
-      ],
-    });
-    tokensController = new TokensController({
```

Comment:
Yikes, so many controllers. Really glad to see this get deleted!

Reply:
Well, I didn't want to go so far as to remove ApprovalController — that's still there. But, the TransactionController tests don't use a real ApprovalController, so we could copy the patterns we have there, or extract the helper functions so that we can reuse them here.